src/solver/gemm_bwd.cpp

```diff
@@ -228,9 +182,10 @@ size_t GemmBwd1x1_stride2::GetWorkspaceSize(const ExecutionContext& context,
     const auto dy_t_size  = dyDesc.GetElementSize() * GetTypeSize(dyDesc.GetType());
     const auto gemm_trans = dx_t_size + dy_t_size;
 
-    if(gemm_trans > MAX_MEM_ALLOC_SZ)
+    if(gemm_trans > gemm::MaxMemAllocSz(handle, problem))
     {
-        MIOPEN_LOG_I2(gemm_trans << " > " << MAX_MEM_ALLOC_SZ);
+        MIOPEN_LOG_I2("GemmBwd1x1_stride2: " << gemm_trans << " > "
+                                             << gemm::MaxMemAllocSz(handle, problem));
         return 0;
     }
     return gemm_trans;
```

Comment:
What do you think?
`gemm_common.hpp:`
```C++
size_t GetMaxPossibleMem(size_t requested_size, auto handle, auto problem, std::string_view solver_name)
{
    auto max_possible_size = gemm::MaxMemAllocSz(handle, problem);
    if(requested_size > max_possible_size)
    {
        MIOPEN_LOG_I2(solver_name << ": " << requested_size << " > " << max_possible_size);
        return 0;
    }
    return requested_size;
}
```

```C++
return GetMaxPossibleMem(gemm_trans, handle, problem, "GemmBwd1x1_stride2");
```

Reply:
@CAHEK7 I can do that if you insist (but please do not ;)

Reply:
@CAHEK7 What I actually think is: returning 0 is a gemm-specific temporary hack that I would like to eliminate. In IsApplicable() we should get WS size and return false if it exceeds allowed max. I will do this while working on #2808.

Reply:
@CAHEK7 ...and I think that for fixing 2808, IsApplicable should also include imitation of rocblas launch to (detect overflows), while this gemm::MaxMemAllocSz() will de-evolute down to handle.GetMaxMemoryAllocSize().